Cpp/odin-views/app.cpp

```diff
@@ -155,12 +149,16 @@ namespace {
             auto const identity_id = fostlib::coerce<fostlib::string>(fed_response_data["identity"]["id"]);
             odin::create_user(cnx, ref, identity_id);
 
-            if ( odin::is_module_enabled(cnx, "opts/full-name") ) {
+            if ( odin::is_module_enabled(cnx, "opts/full-name")
+                && !fed_response_data["identity"]["full_name"].isnull()
+                && fed_response_data["identity"]["full_name"] != "" ) {
                 odin::set_full_name(cnx, ref, identity_id,
                     fostlib::coerce<fostlib::string>(fed_response_data["identity"]["full_name"]));
             }
 
-            if ( odin::is_module_enabled(cnx, "opts/email") ) {
+            if ( odin::is_module_enabled(cnx, "opts/email")
+                && !fed_response_data["identity"]["email"].isnull()
+                && fed_response_data["identity"]["email"] != "" ) {
```

Comment:
Probably you want to do the local JSON tests before the module tests as the module tests will hit the database, which we can save if the data isn't present anyway.